pta-intelligent-search-metadata-extractor/src/main/java/fi/maanmittauslaitos/pta/search/metadata/json/CKANMetadataExtractorConfigurationFactory.java

```diff
@@ -168,42 +171,67 @@ public DocumentProcessingConfiguration createMetadataDocumentProcessingConfigura
 		bboxExtractor.setDefaultValue(DEFAULT_BOUNDING_BOX_FOR_CKAN_METADATA);
 		extractors.add(bboxExtractor);
 
-		return configuration;
-	}
+		// download links
+		extractors.add(createCustomListJsonPathExtractor(
+				ResultMetadataFields.DOWNLOAD_LINKS,
+				new DownloadLinksCkanCustomExtractor(),
+				"$.resources[0].['url', 'name', 'description']"
+		));
 
-	@Override
-	public DocumentProcessor createMetadataDocumentProcessor() throws ParserConfigurationException {
-		DocumentProcessingConfiguration configuration = createMetadataDocumentProcessingConfiguration();
-		return getDocumentProcessorFactory().createJsonProcessor(configuration);
-	}
+		// CKAN custom field used to extract creation date
+		FieldExtractorConfigurationImpl creationDateExtractor = (FieldExtractorConfigurationImpl) createFirstMatchingJsonPathExtractor(
+				ResultMetadataFields.CKAN_CREATION_DATE,
+				"$.resources[0].created", "$.metadata_created"
+		);
+		creationDateExtractor.setTrimmer(dStamp -> dStamp != null ? dStamp.trim().split("\\.")[0] : null);
+		extractors.add(creationDateExtractor);
 
-	private FieldExtractorConfiguration createJsonPathExtractor(String field, FieldExtractorType type, String jsonPath) {
-		FieldExtractorConfigurationImpl ret = new FieldExtractorConfigurationImpl();
-		ret.setField(field);
-		ret.setType(type);
-		ret.setQuery(jsonPath);
-		return ret;
-	}
 
-	private FieldExtractorConfiguration createCustomJsonPathExtractor(String field, CustomExtractor extractor, String jsonPath) {
-		FieldExtractorConfigurationImpl ret = (FieldExtractorConfigurationImpl) createJsonPathExtractor(field,
-				FieldExtractorType.CUSTOM_CLASS, jsonPath);
-		ret.setCustomExtractor(extractor);
-		return ret;
+		// organizations
+		extractors.add(createCustomListJsonPathExtractor(
+				ResultMetadataFields.ORGANISATIONS_RESOURCE,
+				new ResponsiblePartyCKANCustomExtractor(),
+				"$.['reporting_organization','reporting_person_email']"
+
+		));
+
+		extractors.add(createCustomListJsonPathExtractor(
+				ResultMetadataFields.ORGANISATIONS_METADATA,
+				new ResponsiblePartyCKANCustomExtractor(),
+				"$.['author', 'author_email']"
+
+		));
+
+		// we use a dummy key so the fields are parsed in the same format as the default org config and thus we can use the default
```

Comment:
Add a TODO or FIXME explaining the hack and noting that this should be taken care of at some point

Reply:
Added a FIXME with a short desc